src/System/include/BipedalLocomotion/System/NamedTuple.h

```diff
@@ -0,0 +1,391 @@
+    constexpr typename std::enable_if<I == sizeof...(Params),
+                                      const std::size_t>::type static get_element_index() noexcept
+    {
+        return error;
+    }
+
+    /**
+     * Get the index of the element given the hash.
+     * @return The index of  the element associated to the given hash.
+     * @note This function is resolved at compile time. So if the hash is not found an error flag
+     * will be returned.
+     */
+    template <std::size_t I = 0, hash_type Hash>
+    constexpr typename std::enable_if<(I < sizeof...(Params)),
+                                      const std::size_t>::type static get_element_index() noexcept
+    {
+        using elementType = typename std::tuple_element<I, std::tuple<Params...>>::type;
+        if constexpr (Hash == elementType::hash)
+        {
+            return I;
+        }
+        return get_element_index<I + 1, Hash>();
+    }
+
+    static constexpr std::size_t error = -1;
+};
+
+/**
+ * Creates a named_tuple object, deducing the target type from the types of arguments.
+ * @params args zero or more arguments to construct the named_tuple from
+ * @return A named_tuple object containing the given values, created as if by
+ * `named_tuple<Args...>(std::forward<Args>(args)...).`
+ */
+template <typename... Args> constexpr auto make_named_tuple(Args&&... args)
+{
+    return named_tuple<Args...>(std::forward<Args>(args)...);
+}
+
+namespace literals
+{
+
+/**
+ * Forms an hash from a char*
+ * @note To use the literal operator please refer to the following code
+ * \code{.cpp}
+ * using namespace BipedalLocomotion::System::literals;
+ *
+ * constexpr BipedalLocomotion::System::hash_type hash = "foo"_h;
+ * \endcode
+ */
+constexpr hash_type operator"" _h(const char* tag, std::size_t s)
+{
+    return ::BipedalLocomotion::System::detail::sid_hash(tag);
+}
+} // namespace literals
+
+} // namespace System
+} // namespace BipedalLocomotion
+
+
+namespace std
+{
+
+/**
+ * Extracts the Ith element from the named_tuple. I must be an integer value in `[0,
+ * sizeof...(Types))`.
+ * @param t the named_tuple
+ * @return A reference to the selected element of t.
+ */
+template <std::size_t I, typename... Params>
+typename std::tuple_element<
+    I,
+    typename ::BipedalLocomotion::System::named_tuple<Params...>::UnderlineTuple>::type&
+get(::BipedalLocomotion::System::named_tuple<Params...>& t) noexcept
+{
+    return t.template get<I>();
+}
+
+/**
+ * Extracts the Ith element from the named_tuple. I must be an integer value in `[0,
+ * sizeof...(Types))`.
+ * @param t the named_tuple
+ * @return A reference to the selected element of t.
+ */
+template <std::size_t I, typename... Params>
+const typename tuple_element<
+    I,
+    typename ::BipedalLocomotion::System::named_tuple<Params...>::UnderlineTuple>::type&
+get(const ::BipedalLocomotion::System::named_tuple<Params...>& t) noexcept
+{
+    return t.template get<I>();
+}
```

Comment:
What about the hash version?

Reply:
I meant to have a ``get_from_hash`` version too. Are there issues even if we keep using the integral type for the hash?

Reply:
in the `std` namespace?

Reply:
Oh right, that wouldn't be good. Nevermind!